### ISSUE TITLE
Setting up Tests for the System

### ISSUE THREAD

**Issue by roseteague** [jarvist/TheDancer.jl#4]
@jarvist I'm starting to think about what tests would be appropriate. So far I'm considering an AdiabaticPropagation test and UnitaryPropagation test using a basic 3 site system. 

Do you think it would be worth including tests for the individual functions like UpdateDipole and UpdateEnergy as well?

**Comment by jarvist** [comment]
Yes, definitely - starting with microscopic tests for update steps, test that things meant to be unitary are actually, etc. is a very good idea.
